swift/WorkflowUI/Sources/ViewControllerDescription/DescribedViewController.swift

```diff
@@ -19,51 +19,60 @@
 import UIKit
 
 
-internal final class AnyScreenViewController: ScreenViewController<AnyScreen> {
+public final class DescribedViewController: UIViewController {
 
-    typealias WrappedViewController = UIViewController & UntypedScreenViewController
+    var currentViewController: UIViewController
 
-    private (set) internal var currentViewController: WrappedViewController
-
-    required init(screen: AnyScreen, viewRegistry: ViewRegistry) {
-        currentViewController = screen.makeViewController(from: viewRegistry)
-        super.init(screen: screen, viewRegistry: viewRegistry)
-
-        addChild(currentViewController)
-        currentViewController.didMove(toParent: self)
+    public init(description: ViewControllerDescription) {
+        currentViewController = description.buildViewController()
+        super.init(nibName: nil, bundle: nil)
     }
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        view.addSubview(currentViewController.view)
+    public convenience init<S: Screen>(screen: S) {
+        self.init(description: screen.viewControllerDescription)
     }
 
-    override func viewDidLayoutSubviews() {
-        super.viewDidLayoutSubviews()
-        currentViewController.view.frame = view.bounds
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) is unavailable")
     }
 
-    override func screenDidChange(from previousScreen: AnyScreen) {
-        super.screenDidChange(from: previousScreen)
-
-        if type(of: screen.wrappedScreen) == currentViewController.screenType {
-            currentViewController.update(untypedScreen: screen.wrappedScreen)
+    public func update(description: ViewControllerDescription) {
+        if description.canUpdate(viewController: currentViewController) {
+            description.update(viewController: currentViewController)
         } else {
-            currentViewController.willMove(toParent: nil)
             if isViewLoaded {
+                currentViewController.willMove(toParent: nil)
                 currentViewController.view.removeFromSuperview()
+                currentViewController.removeFromParent()
             }
-            currentViewController.removeFromParent()
-
-            currentViewController = screen.makeViewController(from: viewRegistry)
-            addChild(currentViewController)
+            currentViewController = description.buildViewController()
             if isViewLoaded {
+                addChild(currentViewController)
                 view.addSubview(currentViewController.view)
+                currentViewController.view.frame = view.bounds
```

Comment:
Do we need to set the frame here? It's set in `viewDidLayoutSubviews`, also, we're not setting it in `viewDidLoad`, curious the heuristics of when to set frame as I came across this in the split screen container work.

Reply:
We either need to set the frame or `view.setNeedsLayout()`. I guess I opted for the former